File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
    if n<= 2 :
        return 1 
    
    cnt = 0
    
    while True :
        if n == 2 or n == 1 :
            cnt+= 1
            return cnt
        if n %2 == 0 :
            n = n // 2 
            continue
        else :
            n = n-1 
            cnt += 1

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int ans = 0;
        
        while (n >= 1)
        {
            if (n ==1)
            {
                ans +=1;
                n = 0 ;
                continue;
            }
            
            if (n%2 ==0)
            {
                n/=2;
            } else {
                ans +=1;
                n-=1;
            }
        }

        return ans;
    }
}